### ISSUE TITLE
Blazor Server AuthenticationStateProvider.AuthenticationStateChanged is not invoked when user logs in/logout

### ISSUE THREAD

**Issue by MilesFan** [dotnet/aspnetcore#42514]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In a Blazor server project, the event subscribed to AuthenticationStateProvider.AuthenticationStateChanged is not invoked when a user logs in/out.

### Expected Behavior

When a user logs in/out, AuthenticationStateProvider.AuthenticationStateChanged should invoke the event for further processing.

### Steps To Reproduce

To reproduce the issue in a minimal example, there is no changes to the whole project besides the steps below.

1, Open Visual Studio 2022, create a new **Blazor Server** application with **Individual Account** authentication.
2, Verify the **AuthenticationStateProvider** service has been registered as scoped in **Programs.cs**. (no change, just to check)
3, Replace everything in the home page **Index.razor** with the following code:
```
@page "/"
@inject AuthenticationStateProvider authenticationStateProvider
@code{
    protected override async Task OnInitializedAsync()
    {
        authenticationStateProvider.AuthenticationStateChanged += onAuthenticationStateChanged;
    }
    public async void onAuthenticationStateChanged(Task<AuthenticationState> authState)
    {
        // To-Do
    }
}
```
4, Run **update-database** in Package Manager Console;
5, Run the application, register*, login, and go back to home page (as in step 3);
6, Open another browser tab for the home page and click logout;
7, Test failed, **onAuthenticationStateChanged** in home page is not invoked.

*To simply this example, email verification is skipped by modifying the database (emailconfirmed=>true).

### .NET Version

6.0.300

**Comment by javiercn** [comment]
@MilesFan thanks for contacting us.

This is by design. By default, the identity of the user gets established when the connection is created and stays the same for the lifetime of the websocket connection.

You can implement a RevalidatingAuthenticationStateProvider to update the user identity. See https://gist.github.com/SteveSandersonMS/6426aad14a68621dc7b0bcea8e43a900 for details.

**Comment by MilesFan** [comment]
@javiercn Thanks for explaining what the RevalidatingAuthenticationStateProvider can do.

But I still have confusion, or maybe I do not understand correctly:

1, Before the **RefreshInterval** is reached in **RevalidatingAuthenticationStateProvider**, the previous login can be still effective somewhere (another browser tab for example). Isn't it a security concern? I guess shortening the interval to a few seconds may lead to performance issues in the **RefreshLoop**.

2, If  by design **AuthenticationStateProvider.AuthenticationStateChanged** will not work like I expected, what is it intended to do?

**Comment by MilesFan** [comment]
After a whole day digging into some blogs and stackoverflow posts, I come up the following non-official explanations that basically determine my code would just fail:

1, **AuthenticationStateProvider** is registered as Scoped service, as it should be.
2, In Blazor Server, scoped service is NOT shared between different browser tabs.
3, In Blazor Server, scoped service is NOT persisted across page refreshes.

In short I get different instances of **AuthenticationStateProvider** on different browser tabs.

So I guess I have to give up this seemingly convinient approach for my case.
